src/codegen/compile.ml

```diff
@@ -10941,31 +11382,99 @@ and compile_prim_invocation (env : E.t) ae p es at =
 
 
   | OtherPrim "popcnt8", [e] ->
```

Comment:
Maybe, we should internally rename this to `popcntNat8` in symmetry with the new specialized `popcntInt8` etc.

Reply:
Yes, I didn't want to do that in case some uses the prim directly (though that's strictly not supported).
Even better would be if we allowed prims to have type parameters, so we could resolve overloading more cleanly, but that's maybe work for a separate PR.